public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -537,37 +555,47 @@ export class CloudWatchDatasource
     const requestParams = {
       from: range.from.valueOf().toString(),
       to: range.to.valueOf().toString(),
-      queries: queryParams.map((param: any) => ({
-        refId: 'A',
-        intervalMs: 1, // dummy
-        maxDataPoints: 1, // dummy
-        datasource: this.getRef(),
-        type: 'logAction',
-        subtype: subtype,
-        ...param,
-      })),
+      queries: queryParams.map(
+        (param: GetLogEventsRequest | StartQueryRequest | DescribeLogGroupsRequest | GetLogGroupFieldsRequest) => ({
+          refId: (param as StartQueryRequest).refId || 'A',
+          intervalMs: 1, // dummy
+          maxDataPoints: 1, // dummy
+          datasource: this.getRef(),
+          type: 'logAction',
+          subtype: subtype,
+          ...param,
+        })
+      ),
     };
 
     if (options.makeReplacements) {
-      requestParams.queries.forEach((query) => {
-        const fieldsToReplace: Array<
-          keyof (GetLogEventsRequest & StartQueryRequest & DescribeLogGroupsRequest & GetLogGroupFieldsRequest)
-        > = ['queryString', 'logGroupNames', 'logGroupName', 'logGroupNamePrefix'];
-
-        for (const fieldName of fieldsToReplace) {
-          if (query.hasOwnProperty(fieldName)) {
-            if (Array.isArray(query[fieldName])) {
-              query[fieldName] = query[fieldName].map((val: string) =>
-                this.replace(val, options.scopedVars, true, fieldName)
-              );
-            } else {
-              query[fieldName] = this.replace(query[fieldName], options.scopedVars, true, fieldName);
+      requestParams.queries.forEach(
+        (query: GetLogEventsRequest | StartQueryRequest | DescribeLogGroupsRequest | GetLogGroupFieldsRequest) => {
+          const fieldsToReplace: Array<
+            keyof (GetLogEventsRequest & StartQueryRequest & DescribeLogGroupsRequest & GetLogGroupFieldsRequest)
+          > = ['queryString', 'logGroupNames', 'logGroupName', 'logGroupNamePrefix'];
+
+          const anyQuery: any = query;
+          for (const fieldName of fieldsToReplace) {
+            if (query.hasOwnProperty(fieldName)) {
+              if (Array.isArray(anyQuery[fieldName])) {
+                anyQuery[fieldName] = anyQuery[fieldName].map((val: string) =>
+                  this.replace(val, options.scopedVars, true, fieldName)
+                );
+              } else {
+                anyQuery[fieldName] = this.replace(anyQuery[fieldName], options.scopedVars, true, fieldName);
+              }
             }
           }
+          // TODO: seems to be some sort of bug that we don't really send region with all queries. This means
```

Comment:
Is there an issue for this?

Reply:
Didn't see any. Maybe nobody noticed yet. It's just a result of adding the correct types but didn't want to add too much additional stuff to this PR.

Reply:
https://github.com/grafana/grafana/issues/41689